File: VB/Services/ReportResolver.vb
Imports DevExpress.XtraReports.UI
Imports DevExpress.XtraReports.Web.WebDocumentViewer
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web

Namespace T132094.Services
	Public Class CustomWebDocumentViewerReportResolver
		Implements IWebDocumentViewerReportResolver

		Public Sub New()
		End Sub
		Public Function Resolve(ByVal reportUniqueName As String) As XtraReport Implements IWebDocumentViewerReportResolver.Resolve
			Select Case reportUniqueName
				Case "DefaultReport"
					Return New DefaultReport()
				Case "Categories"
					Return New CategoriesReport()
				Case "Products"
					Return New ProductsReport()
				Case "Orders"
					Return New OrdersReport()
				Case Else
					' Try to create a report using the fully qualified type name.
					Dim t As Type = Type.GetType(reportUniqueName)
					Return If(GetType(XtraReport).IsAssignableFrom(t), DirectCast(Activator.CreateInstance(t), XtraReport), Nothing)
			End Select
		End Function
	End Class
End Namespace

File: CS/Services/ReportResolver.cs
using DevExpress.XtraReports.UI;
using DevExpress.XtraReports.Web.WebDocumentViewer;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace T132094.Services {
    public class CustomWebDocumentViewerReportResolver : IWebDocumentViewerReportResolver {
        public CustomWebDocumentViewerReportResolver() { }
        public XtraReport Resolve(string reportUniqueName) {
            switch (reportUniqueName) {
                case "DefaultReport":
                    return new DefaultReport();
                case "Categories":
                    return new CategoriesReport();
                case "Products":
                    return new ProductsReport();
                case "Orders":
                    return new OrdersReport();
                default:
                    // Try to create a report using the fully qualified type name.
                    Type t = Type.GetType(reportUniqueName);
                    return typeof(XtraReport).IsAssignableFrom(t) ?
                        (XtraReport)Activator.CreateInstance(t) :
                        null;
            }
        }
    }
}
